wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/environment/server/CheEnvironmentEngine.java

```diff
@@ -887,47 +889,86 @@ Instance startMachine(LineConsumer machineLogger,
         }
-        return serviceWithCorrectSource;
+        return serviceWithNormalizedSource;
     }
 
-    private Machine getMachineWithCorrectSource(MachineImpl machine, MachineSource machineSource) {
-        Machine machineWithCorrectSource = machine;
+    private Machine getMachineWithNormalizedSource(MachineImpl machine, MachineSource machineSource) {
+        Machine machineWithNormalizedSource = machine;
         if (machineSource != null) {
-            machineWithCorrectSource = MachineImpl.builder()
-                                                  .fromMachine(machine)
-                                                  .setConfig(MachineConfigImpl.builder()
-                                                                              .fromConfig(machine.getConfig())
-                                                                              .setSource(machineSource)
-                                                                              .build())
-                                                  .build();
+            machineWithNormalizedSource = MachineImpl.builder()
+                                                     .fromMachine(machine)
+                                                     .setConfig(MachineConfigImpl.builder()
+                                                                                 .fromConfig(machine.getConfig())
+                                                                                 .setSource(machineSource)
+                                                                                 .build())
+                                                     .build();
+
+        }
+        return machineWithNormalizedSource;
+    }
+
+    /**
+     * Replaces linked to this service's name with container name which represents the service in links section.
+     * The problem is that a user writes names of other services in links section in compose file.
+     * But actually links are constraints and they values should be names of containers (not services) to be linked.
+     * <br/>
+     * For example: serviceDB:serviceDbAlias -> container_1234:serviceDbAlias <br/>
+     * If alias is omitted then service name will be used.
+     *
+     * @param serviceToStart
+     *         service which is starting now
+     * @param services
+     *         all services in environment
+     */
+    @VisibleForTesting
+    void replaceServiceToContainerNameInLinks(CheServiceImpl serviceToStart, Map<String, CheServiceImpl> services) {
+        final List<String> containerLinks = serviceToStart.getLinks();
+        for (int i = 0;  i < containerLinks.size(); i++) {
+            String link = containerLinks.get(i);
+            String serviceName;
+            String serviceAlias;
+            int colonIndex = link.indexOf(':');
+            if (colonIndex != -1) {
+                serviceName = link.substring(0, colonIndex);
+                serviceAlias = link.substring(colonIndex + 1);
```

Comment:
If user provide link with empty alias, like `serviceDb:`, than `CheEnvironmentValidator` will throw an exception and a user will be notified about syntax error. But some logs will not be superfluous, I think.